StRoot/StSpinPool/StTriggerFilterMaker/StTriggerFilterMaker.cxx

```diff
@@ -21,39 +22,85 @@ Int_t StTriggerFilterMaker::Init() {
 }
 
 Int_t StTriggerFilterMaker::Make() {
-    StTriggerIdCollection* trgIdColl;
-    StEvent* event= (StEvent*)GetInputDS("StEvent");
-    if(event){
-	trgIdColl = event->triggerIdCollection();
-    }else{
-	if(StMuDst::event()){
-	    trgIdColl = event->triggerIdCollection();
-	}else{
-	    LOG_INFO << "No StEvent nor Mudst" << endm;
-	}	
-    }
+  StEvent* event= (StEvent*)GetInputDS("StEvent");
+  if(event){
+    StTriggerIdCollection* trgIdColl = event->triggerIdCollection();
+    const StTriggerData* trgdata = event->triggerData();
     if(trgIdColl){
-	if(mPrint){
-	    vector<unsigned int> ids = trgIdColl->nominal()->triggerIds();
-	    LOG_INFO << "Offline Trigger Id = ";
-	    for(unsigned i=0; i<ids.size(); ++i) LOG_INFO << ids[i] << " ";
-	}
-	for(unsigned i=0; i<mVetoTriggers.size(); ++i) {
-	    if(trgIdColl->nominal()->isTrigger(mVetoTriggers[i])) {
-		if(mPrint)  LOG_INFO << " Veto" << endm;
-		return kStSkip;
-	    }
+      if(mPrint){
+	vector<unsigned int> ids = trgIdColl->nominal()->triggerIds();
+	LOG_INFO << "Offline Trigger Id = ";
+	for(unsigned i=0; i<ids.size(); ++i) LOG_INFO << ids[i] << " ";
+      }
+      for(unsigned i=0; i<mVetoTriggers.size(); ++i) {
+	if(trgIdColl->nominal()->isTrigger(mVetoTriggers[i])) {
+	  if(mPrint)  LOG_INFO << " Vetoed" << endm;
+	  return kStSkip;
 	}
-	for(unsigned i=0; i<mGoodTriggers.size(); ++i) {
-	    if(trgIdColl->nominal()->isTrigger(mGoodTriggers[i])) {
-		if(mPrint)   LOG_INFO << " Accept" << endm;
-		return kStOk;
+      }
+      for(unsigned i=0; i<mGoodTriggers.size(); ++i) {
+	if(trgIdColl->nominal()->isTrigger(mGoodTriggers[i])) {
+	  
+	  if(trgdata && mTofUpperLimit>0){
+	    unsigned int tof = trgdata->tofMultiplicity();
+	    if(tof < mTofUpperLimit){
```

Comment:
It is not clear how this relates to trigger filtering. What trigger is this supposed to emulate? Would it make sense to have a separate StTofMultFilterMaker?

Reply:
This emulates peripheral trigger which did not need to be implemented at trigger level since we were taking MinBias data with no pre-scale. Yet we could have make one, and push to another steam, which would be more convenient for us, at a cost of more disk & HPSS tape space and extra work by experts to set up & longer meetings to get approval for a very short (~3days) of running. So we decided not to.
  I can make it to another maker. But it is also natural, convenient and less work/less complicated macros at end, to pout it here.

Reply:
> But it is also natural, convenient and less work/less complicated macros at end, to pout it here.

I don't see your point, but okay.